File: byteBank-herdado-conta/src/br/com/sgsistemas/java/trainee/modelo/ContaCorrente.java
package br.com.sgsistemas.java.trainee.modelo;

//new br.com.sgsistemas.java.trainee.modelo.ContaCorrente()
public class ContaCorrente extends Conta implements Tributavel{

	public ContaCorrente(int agencia, int numero) {
		super(agencia, numero);
	}
	
	@Override
	public void saca(double valor) /*throws br.com.sgsistemas.java.trainee.modelo.SaldoInsuficienteException*/ {
		double valorASacar = valor + 0.2;
		 super.saca(valorASacar);
	}

	@Override
	public void deposita(double valor) {
        super.saldo += valor;
    }

	@Override
	public double getValorImposto() {
		return super.saldo * 0.01;
	}
}

File: JavaScript/byteBank/Conta/ContaCorrente.js
import { Cliente } from "../Cliente.js"
import { Conta } from "./Conta.js"

export class ContaCorrente extends Conta{

    static numeroDeContas = 0

    constructor (agencia,cliente){
       super(agencia,cliente,0)
        ContaCorrente.numeroDeContas++
    }

    sacar(valor){
       const taxa = 1.2
        super._sacar(valor,taxa)
    }
    
  
}
